---
title: Getting started with Spatial Transcriptomics
output:
  html_document:
    toc: true
    toc_float:
      toc_collapsed: true
    toc_depth: 3
    number_sections: false
css: style.css
---
```{r echo = FALSE, show = FALSE}
knitr::opts_chunk$set(number_sections = FALSE)
```

## Get workshop files

<input type="checkbox"> Download the files for this activity clicking here: https://github.com/fhdsl/ITN_Workshops_2024/archive/refs/heads/main.zip   
<input type="checkbox"> Put this file on your desktop so it is easily findable.  
<input type="checkbox"> Double click the zip file (or right click and choose "unzip" or "decompress" to unzip the file.  
<input type="checkbox"> Open up your activity files you downloaded so we can see what's here.  

### Get familiar with the data we have

We should see we have four samples, a metadata file, as well as a PDF of the manuscript that describes these data.

Each sample's folder has an h5 file, and a spatial folder.

- sample_093d
  - GSM6433590_093D_filtered_feature_bc_matrix.h5
  - spatial
    - GSM6433590_093D_scalefactors_json.json
    - GSM6433590_093D_tissue_hires_image.png
    - GSM6433590_093D_tissue_positions_list.csv
- sample_396a
  ...
- sample_396c
  ...
- sample_397d
  ...

The README describes these samples:

> The original data set results from triple-negative breast cancer (TNBC) tumor biopsies collected from a diverse patient cohort. Patients underwent different therapeutic regimes. The tumor biopsies were profiled with 10X Visium. The data set presented here is a subset of the original cohort.

Additionally the metadata describe which samples came from which patients with which treatments which looks like this:

|samplename|patient|therapy|
|:----------|:-------|:-------|
|sample_093d|patient_12|none
|sample_396a|patient_19|pembrolizumab
|sample_396c|patient_20|adriamycin
|sample_397d|patient_22|taxotere

While 4 samples are provided, this workshop will only use 2 of the 4 samples. The rest could be used with this activity walkthrough after the workshop.  

## Create an account with SpatialGE

<input type="checkbox"> Go to https://spatialge.moffitt.org/  
<input type="checkbox"> Click on "Sign Up" in the upper right corner.  

## Starting a new project

<input type="checkbox"> Click the blue "New Project" button.    
<input type="checkbox"> For `What spatial transcriptomics platform are you using for this project?` choose `Visium` -- this is the type of data our example data are.  
<input type="checkbox"> Make your own project name and description that's sensible. Could be something related to the workshop "ITN workshop".    
<input type="checkbox"> Then click "Create".  

## Uploading the dataset

_For each sample we will repeat the following steps to upload each sample's set of files._

### Uploading one sample's data

<input type="checkbox"> For `Sample Name` put the ID indicating on the folder, e.g. `sample_093d`      
<input type="checkbox"> For the `Gene expression` box upload the `.h5` file e.g. `GSM6433590_093D_filtered_feature_bc_matrix.h5`. You can upload files by dragging and dropping or by clicking on them to navigate.   
<input type="checkbox"> For the `Coordinates` box upload the `.csv` file e.g. `GSM6433590_093D_tissue_positions_list.csv`.   
<input type="checkbox"> For the `Tissue image` box upload the `.png` file e.g. `GSM6433590_093D_tissue_hires_image.png`.   
<input type="checkbox"> For the `Scale factor` box upload the `.json` file e.g. `GSM6433590_093D_scalefactors_json.json`. 

<input type="checkbox"> Once the above steps are done click the green `Import Sample`.

_You can use this checklist to keep track as you upload and follow the steps for each sample._

<input type="checkbox"> `sample_093d` data entered  
<input type="checkbox"> `sample_396c` data entered  

#### {.click_to_expand_block}
<details><summary>Click here for additional datasets for post-workshop walkthrough.</summary>

<input type="checkbox"> `sample_396a` data entered  
<input type="checkbox"> `sample_397d` data entered

</details>

####
Now [return to the beginning of these steps](#uploading-one-samples-data) to repeat the same steps for the other sample.

### Adding metadata

<input type="checkbox"> Now click `Add metadata manually`.    
<input type="checkbox"> Click `Add new metadata column`. Add a column named `patient`.   
<input type="checkbox"> Click `Add new metadata column` again. Add a column named `therapy`.     

You can reference the `example_clinical.csv` file's contents to add these data for each sample:

|samplename|patient|therapy|
|:----------|:-------|:-------|
|sample_093d|patient_12|none
|sample_396c|patient_20|adriamycin

<input type="checkbox"> Add this `sample_093d` corresponding `patient` and `therapy` information.    
<input type="checkbox"> Add this `sample_396c` corresponding `patient` and `therapy` information.   

#### {.click_to_expand_block}
<details><summary>Click here for additional datasets for post-workshop walkthrough.</summary>

|samplename|patient|therapy|
|:----------|:-------|:-------|
|sample_396a|patient_19|pembrolizumab
|sample_397d|patient_22|taxotere

<input type="checkbox"> Add this `sample_396a` corresponding `patient` and `therapy` information.  
<input type="checkbox"> Add this `sample_397d` corresponding `patient` and `therapy` information.   

</details>

####

### After you've entered the data and metadata:  

<input type="checkbox">  You will not be able to edit this (unless you delete the sample completely) after you click `Import Data`. So make sure everything is as you intend and then click `Import Data`. This will take a little bit of time. Note you can have it send you an email instead of waiting on the page.  

## Filtering your data

*Note that these values for the filtering steps are suggested for this data and may not be appropriate for all future use/datasets.*

<input type="checkbox"> Go to the "Filter data" tab.  
<input type="checkbox"> Click "Filter spots/cells".    
<input type="checkbox"> Enter the **minimum** number of counts a spot needs to have to be kept in the data set. In this case, 2000 will be input.  
<input type="checkbox"> Enter the **minimum** number of genes a spot needs to have to be kept in the data set. In this case, 500 will be input.  
<input type="checkbox"> Click the "Mitochondrial genes (^MT-)" field to filter spots by mitochondrial gene content. Keep in mind that some ST platforms do not quantify mitochondrial genes.  
<input type="checkbox"> Enter the **maximum** percentage of mitochondrial counts. Use 20% in this case.  
<input type="checkbox"> Now, to filter out genes, click "Filter genes".  
<input type="checkbox"> Filter out genes with **less than** 2000 counts.  
<input type="checkbox"> Filter out genes expressed in **less than** 20 spots.  
<input type="checkbox"> Once you have all the filter settings as you'd like click the blue "APPLY FILTER" button.  

### Data visualization of filtering

<input type="checkbox"> Click "Violin plots" to visualize count distribution after filtering.    
<input type="checkbox"> Currently, "total_counts" and "total_genes" per spot can be visualized.    
<input type="checkbox"> When changing the variable to plot, click the blue "GENERATE PLOTS" button to update.    

## Normalize Data

<input type="checkbox"> Click the "Normalize data" tab.  
<input type="checkbox"> Click "Use SCTransform" to apply Seurat's normalization method.  
<input type="checkbox"> Click the blue "NORMALIZE DATA" to start normalization.  
<input type="checkbox"> The distribution of counts per spot for a given gene can also be plotted. For example, TP53.  
<input type="checkbox"> Click "GENERATE PLOTS" to show the number of TP53 counts per spot.  


## Visualization

### Quilt plot

<input type="checkbox">  Click `Quilt plot` to visualize the total number of genes or counts per spot and their spatial context.  
<input type="checkbox"> Select `total_counts`.  
<input type="checkbox">  Select one sample underneath the `First sample` dropdown menu.  
<input type="checkbox"> And select a second sample to compare to underneath the `Second sample` dropdown menu.  
<input type="checkbox"> Click blue "GENERATE PLOTS" button to create the plot.  

### Gene comparison visualization

<input type="checkbox"> Click the `Visualization` module on the left side menu.   
<input type="checkbox"> You can search for your favorite gene in the `Search and select genes` menu. For this example query and click "IGKC".  
<input type="checkbox"> Also query and click "FN1" gene.  
<input type="checkbox"> Lastly, also query and click `C1QA`.  
<input type="checkbox"> Click blue "GENERATE PLOTS" button to create the plot.   

### Expression surface

Alternatively, an "expression surface" can be generated. This is a type of plot where expression values are inferred for the spaces not quantified between spots

<input type="checkbox"> Click the "Expression surface" tab.  
<input type="checkbox"> In the `Search and select genes` menu search and select `IGKC`.  
<input type="checkbox"> Click "ESTIMATE SURFACES" button to create the plot.   

## Spatial Domain Detection

<input type="checkbox"> Click the "Spatial domain detection" on the left side menu.  
<input type="checkbox"> Now in the `Number of domains` slider put 3 to 6 domains will be detected in the samples.  his is how many clusters will attempt to be identified.   
<input type="checkbox"> For `Number of most variable genes to use` choose 3000 with high variation will be used to detect the domains.  
<input type="checkbox"> Finally, click "RUN STCLUST" to find clusters.  
<input type="checkbox"> Explore the results by clicking each `K=` tab.  

Images can be exported in multiple formats.
